src/Kentico.Xperience.Lucene/Services/Implementations/DefaultLuceneTaskProcessor.cs

```diff
@@ -65,9 +100,48 @@ public async Task<int> ProcessLuceneTasks(IEnumerable<LuceneQueueItem> queueItem
                 eventLogService.LogError(nameof(DefaultLuceneTaskProcessor), nameof(ProcessLuceneTasks), ex.Message);
             }
         }
+    }
+
+    private static IEnumerable<string?> GetIdsToDelete(IEnumerable<LuceneQueueItem> deleteTasks) => deleteTasks.Select(queueItem => queueItem.IndexedItemModel.WebPageItemGuid.ToString());
+
+    /// <inheritdoc/>
+    public async Task<Document?> GetDocument(LuceneQueueItem queueItem)
+    {
+        var luceneIndex = IndexStore.Instance.GetIndex(queueItem.IndexName) ?? throw new Exception($"LuceneIndex {queueItem.IndexName} not found!");
+
+        var data = await luceneIndex.LuceneIndexingStrategy.MapToLuceneDocumentOrNull(queueItem.IndexedItemModel);
+
+        if (data is null)
+        {
+            return null;
+        }
+
+        await AddBaseProperties(queueItem.IndexedItemModel, data!);
 
-        return successfulOperations;
+        return data;
     }
 
-    private IEnumerable<string> GetIdsToDelete(IEnumerable<LuceneQueueItem> deleteTasks) => deleteTasks.Select(queueItem => queueItem.Node.DocumentID.ToString());
+    private async Task AddBaseProperties(IndexedItemModel lucenePageItem, Document document)
+    {
+        document.AddStringField(nameof(IndexedItemModel.ClassName), lucenePageItem.ClassName, Field.Store.YES);
+        document.AddStringField(nameof(IndexedItemModel.WebPageItemGuid), lucenePageItem.WebPageItemGuid.ToString(), Field.Store.YES);
+        document.AddStringField(nameof(IndexedItemModel.LanguageCode), lucenePageItem.LanguageCode, Field.Store.YES);
+
+        string url = string.Empty;
+        try
+        {
+            if (lucenePageItem.WebPageItemGuid is not null && lucenePageItem.LanguageCode is not null)
+            {
+                url = (await urlRetriever.Retrieve(lucenePageItem.WebPageItemGuid.Value, lucenePageItem.LanguageCode)).RelativePath;
```

Comment:
Hmm... if content from one web channel is being linked to from another via search results, this would prevent that because it's a relative path.

This should probably be customizable somehow.

Reply:
I have added customization option I.E. developer can add this field as any other field and when that is done the default way is not used. I'm adding this to the documentation as well.